### PR TITLE
Add extra RPC modules (grandpa)

### PR DESCRIPTION
**Pull request by dvdplm**
Adds grandpa and babe RPCs.

### CODE REVIEW

File: bin/node-template/node/src/service.rs [hunk 1]
```diff
@@ -196,6 +197,8 @@ pub fn new_full(mut config: Configuration) -> Result<TaskManager, ServiceError>
 			task_manager: &mut task_manager,
 			transaction_pool: transaction_pool.clone(),
 			rpc_extensions_builder,
+			// TODO: (dp) implement
+			rpsee_builder: Box::new(|_, _| { RpcModule::new(()) }),
```
Comment (niklasad1) [inline bin/node-template/node/src/service.rs:201]: we will take of it in another PR?

File: bin/node-template/node/src/service.rs [hunk 2]
```diff
@@ -12,6 +12,7 @@ use sc_finality_grandpa::SharedVoterState;
 use sc_keystore::LocalKeystore;
 use sc_telemetry::{Telemetry, TelemetryWorker};
 use sp_consensus::SlotData;
+use jsonrpsee_ws_server::RpcModule;
```
Comment (niklasad1) [inline bin/node-template/node/src/service.rs:15]: use it from `jsonrpsee crate` instead?!

File: bin/node-template/node/Cargo.toml [hunk 3]
```diff
@@ -39,6 +39,8 @@ sp-timestamp = { version = "3.0.0", path = "../../../primitives/timestamp" }
 
 # These dependencies are used for the node template's RPCs
 jsonrpc-core = "15.1.0"
+jsonrpsee-ws-server = { git = "https://github.com/paritytech/jsonrpsee", branch = "master" }
+jsonrpsee-types = { git = "https://github.com/paritytech/jsonrpsee", branch = "master" }
```
Comment (niklasad1) [inline bin/node-template/node/Cargo.toml:43]: jsonrpsee-types is unused?

File: bin/node/cli/Cargo.toml [hunk 4]
```diff
@@ -78,6 +79,11 @@ sc-telemetry = { version = "3.0.0", path = "../../../client/telemetry" }
 sc-authority-discovery = { version = "0.9.0",  path = "../../../client/authority-discovery" }
 sc-finality-grandpa-warp-sync = { version = "0.9.0", path = "../../../client/finality-grandpa-warp-sync", optional = true }
 
+sc-finality-grandpa-rpc = { version = "0.9.0", path = "../../../client/finality-grandpa/rpc" }
+jsonrpsee-ws-server = { git = "https://github.com/paritytech/jsonrpsee", branch = "master" }
```
Comment (niklasad1) [inline bin/node/cli/Cargo.toml:83]: same here, would be better just to use `jsonrpsee root` crate

File: client/finality-grandpa/rpc/src/error.rs [hunk 5]
```diff
@@ -45,6 +46,7 @@ pub enum ErrorCode {
 	ProveFinality,
 }
 
+// TODO: remove (?)
```
Comment (niklasad1) [inline client/finality-grandpa/rpc/src/error.rs:49]: we need to add this to jsonrpsee, it's not possible to set application specific error codes.

File: client/finality-grandpa/rpc/src/lib.rs [hunk 6]
```diff
@@ -20,170 +20,116 @@
...
+			grandpa
+				.finality_proof_provider
+				.rpc_prove_finality(block)
+				.map_err(|finality_err| error::Error::ProveFinalityFailed(finality_err))
+				.map_err(to_jsonrpsee_call_error)
+		})?;
+
+		// Returns the block most recently finalized by Grandpa, alongside its justification.
+		module.register_subscription(
+			"grandpa_justifications",
```
Comment (niklasad1) [inline client/finality-grandpa/rpc/src/lib.rs:99]:
```suggestion
			"grandpa_subscribeJustifications",
```

Comment (niklasad1) [reply client/finality-grandpa/rpc/src/lib.rs:99]: https://github.com/paritytech/substrate/blob/master/client/finality-grandpa/rpc/src/lib.rs#L65

**Review by niklasad1 — APPROVED** [review]
looks good nice work, would be good if you could remove the old extensions builder if possible :)

File: bin/node-template/node/src/service.rs (hunk 1 above)
Comment (dvdplm) [reply bin/node-template/node/src/service.rs:201]: Yeah, or, I'm not sure what `rpc_extensions_builder` is here, but in a few places it's just an empty closure just like I do here.

**Comment by dvdplm** [comment]
> remove the old extensions builder if possible :)

Definitely, we'll have to port over all of [these](https://github.com/paritytech/substrate/blob/e6cd6b1240ce741f06b05ad1c59f5399e52e717a/bin/node/rpc/src/lib.rs#L157-L209) first. Working on SyncState now.

**Pull request merged**